api.planx.uk/inviteToPay/createPaymentSendEvents.ts

```diff
@@ -0,0 +1,122 @@
+import { ComponentType } from '@opensystemslab/planx-core/types';
+import { NextFunction, Request, Response } from 'express';
+import { gql } from 'graphql-request';
+import { _admin as $admin } from '../client';
+import { adminGraphQLClient as adminClient } from "../hasura";
+import { createScheduledEvent } from "../hasura/metadata";
+import { getMostRecentPublishedFlow } from '../helpers';
+import { Flow, Node, Team } from '../types';
+
+enum Destination {
+  BOPS = "bops",
+  Uniform = "uniform",
+  Email = "email",
+}
+
+interface CombinedResponse {
+  bops?: Record<string, string>;
+  uniform?: Record<string, string>;
+  email?: Record<string, string>;
+}
+
+// Create "One-off Scheduled Events" in Hasura when a payment request is paid
+const createPaymentSendEvents = async (req: Request, res: Response, next: NextFunction): 
+  Promise<NextFunction | Response | void> => {
+  try {
+    const { payload } = req.body;
+    if (!payload.sessionId) {
+      return next({
+        status: 400,
+        message: `Missing payload data to create payment send events`,
+      });
+    }
+
+    const now = new Date();
+    const combinedResponse: CombinedResponse = {};
+
+    const session = await $admin.getSessionById(payload.sessionId);
+    const publishedFlowData = await getMostRecentPublishedFlow(session.flowId);
+    if (!session || !publishedFlowData) {
+      return next({
+        status: 400,
+        message: `Cannot fetch session or flow data to create payment send events`,
+      });
+    }
+
+    // Find this sessions Send component, determine which "destinations" we need to queue up events for
+    const sendNode: [string, Node] | undefined = Object.entries(publishedFlowData).find(([_nodeId, nodeData]) => nodeData.type === ComponentType.Send);
```

Comment:
Question: do we care that after these events successfully run, we'll have a submitted `lowcal_session` that is marked as `submitted_at` but does not have a `Send` component in its' breadcrumb data? 

My instinct is this is exactly the _expected_ behavior for a session with an associated `payment_request` because it has "left" the graph at the Pay node, but want to checkin/confirm that feels right to others too. If not, we could in theory add this node to the `lowcal_session.data.breadcrumbs` at this stage as well with `auto: true` or similar.

Reply:
That's interesting. I've imagined that breadcrumbs are a temporary data structure used to construct something more canonical (the submission itself, and perhaps the passport as the distillation of the session data). The flows can change (perhaps invalidating breadcrumbs) but the output data (exported submission data/passport) is more stable and that's what we want to persist, ideally in a registry somewhere external, as the canonical submission. The breadcrumbs would ideally be disposable once the submission is complete.